src/main/scala/src/org/aerospiker/Operation.scala

```diff
@@ -43,12 +43,15 @@ trait Operation { self: BaseClient =>
       ).trans
     })
 
-  private def futuring[B, A](e: AerospikeException => B)(f: () => A): EitherT[Future, B, A] =
+  private def futuring[B, A](e: AerospikeException => B)(f: () => Option[A]): EitherT[Future, B, A] =
```

Comment:
By fixing a little verbose pattern matching and literally strange error message:
```scala
  // this code needs `import scalaz._, Scalaz._`
  private def futuring[B, A](e: AerospikeException => B)(f: () => Option[A]): EitherT[Future, B, A] = {
    Future { f() } onComplete {
      case Success(res) => res match {
        case Some(x) => if (x.isInstanceOf[A]) x.right else e(new AerospikeException("not found")).left
        case None => e(new AerospikeException("null record returned")).left
      }
      case Failure(t) => t match {
        case t: AerospikeException => e(t).left
      }
    }
  }
```